llvm/include/llvm/CodeGen/SelectionDAGTargetInfo.h

```diff
@@ -35,6 +35,19 @@ class SelectionDAGTargetInfo {
   SelectionDAGTargetInfo &operator=(const SelectionDAGTargetInfo &) = delete;
   virtual ~SelectionDAGTargetInfo();
 
+  /// Returns true if a node with the given target-specific opcode has
+  /// a memory operand. Nodes with such opcodes can only be created with
+  /// `SelectionDAG::getMemIntrinsicNode`.
+  virtual bool isTargetMemoryOpcode(unsigned Opcode) const { return false; }
```

Comment:
We already have isBinOp / isCommutativeBinOp that do something similar that inherit from TargetLoweringBase - why not put them there instead?

Reply:
Good question.
These methods are going to be autogenerated, and I don't think pulling the autogenerated stuff into TargetLowering would do any good -- the class is already overloaded too much. I'd prefer to do the opposite: move isBinOp / isCommutativeBinOp to SelectionDAGTargetInfo and autogenerate them too.
That is, I would like to make this class look more like TargetInstrInfo, but for SDNodes rather than MachineInstructions.

Reply:
I also see this class as a place for [autogenerated] target-specific queries like RISCV's [hasPassThruOp](https://github.com/llvm/llvm-project/pull/119709/files#diff-c179575972a6aa579b04129e1a96a4c1f4ac0dc457c0f647150eed1b54b5a861R58) (will take some time to scroll to the definition).

Reply:
Cheers - happy for isBinOp / isCommutativeBinOp to be moved over in a future patch